Made gatsby-plugin-netlify-cms work with gatsby v2

emanueleperuffo opened the pull request:
I've made gatsby-plugin-netlify-cms work with gatsby v2.

I've not used webpack-merge because I couldn't get something working in a reasonable amount of time and I didn't have too much time to learn it, so I've ended up using vanilla JavaScript.

ascorbic commented:
Great work. It's worth tagging #6048

shaunbent commented:
Awesome work @emanueleperuffo - I am also blocked from completing my V2 upgrade because of this issue.

m-allanson commented on packages/gatsby-plugin-netlify-cms/package.json:
```diff
@@ -1,15 +1,16 @@
   "name": "gatsby-plugin-netlify-cms",
   "description": "A Gatsby plugin which generates the Netlify CMS single page app",
-  "version": "2.0.0-beta.3",
+  "version": "2.2.0-beta.15",
```
Can you revert this change? Versioning will be handled during the release process.

emanueleperuffo commented:
@m-allanson I've moved to mini-css-extract-plugin and reverted back the version number.
In my previous commit, I increased the version number because npm thinks that v2.0.0-beta.n is a previous version of v2.0.1, which is the version of branch v1.

donysukardi commented:
I tested it. Seems to be working ok. However I'm seeing FOUC when I load the admin page

KyleAMathews commented:
I'll merge and release this now. Could someone investigate the FOUC issue @donysukardi saw and fix that?

steverandy commented:
Tried `gatsby-plugin-netlify-cms@2.0.0-beta.4`. The /admin/ page works on develop but it shows empty page on build.

joshdcuneo commented:
Having the same issue as @steverandy. Looks like maybe the Identity widget isn't being injected when the plugin generates the index.html? 

`<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>Content Manager</title>
  <link href="/cms.ba341ffc331db5549ac2.css" rel="stylesheet"><link href="/11.cms.css" rel="stylesheet"></head>
  <body>
  <script type="text/javascript" src="/cms-f3360073949bf65e6d32.js"></script></body>
</html>`

emanueleperuffo commented:
I see. And I also noticed that the css is injected in 2 separate but
apparently identical files.

On Sat., 7 Jul. 2018, 8:06 am Josh Cuneo, <notifications@github.com> wrote:

> Having the same issue as @steverandy <https://github.com/steverandy>.
> Looks like maybe the Identity widget isn't being injected when the plugin
> generates the index.html?
>
> `
> <title>Content Manager</title> <script type="text/javascript"
> src="/cms-f3360073949bf65e6d32.js"></script> `
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gatsbyjs/gatsby/pull/6292#issuecomment-403191758>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AE2XZekgpTRLKj-d7yEEhkYWagufUpVLks5uEE_-gaJpZM4VCZSV>
> .
>

emanueleperuffo commented:
@steverandy The default export of netlify-identity-widget is undefined for gatsby build, but not for gatsby develop... I don't know how to fix this...

steverandy commented:
I happened to have a similar issue.
I think may need to use `require` to import `netlify-identity-widget`.

I'm not sure why it behaves like this.

emanueleperuffo commented:
Also, I've found this in my production pages (not /admin):
<iframe id="netlify-identity-widget" style="position: fixed; top: 0; left: 0; border: none; width: 100%; height: 100%; overflow: visible; background: transparent; display: none; z-index: 99; " src="about:blank"></iframe>

I think that the code in cms.js goes to the main entry instead of the cms entry in this case...

erquhart commented:
Looking into this (all of the above) now.

emanueleperuffo commented:
I've created a new pr #6341

jlengstorf commented:
Hiya @emanueleperuffo! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs
